### PR TITLE
Improve Build & Check CI to displays test results directly in GitHub

### LINKED ISSUE #7: Improve Build & Check CI to displays test results from directly in GitHub
Add another job with the named `Test-Reporter` within the `build.yml` to do the following:

- Downloads the output of the `build` job (See [actions/download-artifact](https://github.com/actions/download-artifact#download-artifact-v3))
- Use [dorny's test-reporter](https://github.com/dorny/test-reporter) to displays test results from directly in GitHub

This job should only run *after* the `build` job. See: [Worflow syntax - needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds).

### PR DESCRIPTION
**Pull request by hientranea**
Issues's description: https://github.com/u-ways/kotlin-quarkus-realworld-example-app/issues/7

The workflow has passed on my fork project. You can check it here: https://github.com/hientranea/kotlin-quarkus-realworld-example-app/actions/runs/3173264532

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -27,3 +27,21 @@ jobs:
...
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download
+        uses: actions/download-artifact@v3
+        with:
+          name: test-results
+          path: test-results/test/
+      - name: Display structure of downloaded files
```
Comment (u-ways) [inline .github/workflows/build.yml:40]: No need to include this step, can it be removed please? (lines 40 and 41)

**Review by u-ways — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (hientranea) [reply .github/workflows/build.yml]: Thank you for pointing this out. I've just updated it.

**Review by u-ways — DISMISSED** [review]

**Comment by u-ways** [comment]
Looks good now, will merge, thanks for your help :)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@ jobs:
           path: build/test-results/test/
   test-reporter:
     needs: build
+    permissions: read-all|write-all
```
Comment (u-ways) [inline .github/workflows/build.yml:32]: Can you try removing this line and add PR write permission on the top, please?

```yml
    permissions:
      pull-requests: write
```

**Review by u-ways — CHANGES_REQUESTED** [review]

**Review by u-ways — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -7,6 +7,10 @@ on:
   pull_request:
     branches:
       - main
+permissions:
+  pull-requests: write
+  contents: read
```
[the remark this answers is not in the document]
Comment (u-ways) [reply .github/workflows/build.yml:12]: @hientranea I think I will merge this and raise another ticket to try and resolve this issue for new contributors PR.

[the remark this answers is not in the document]
Comment (hientranea) [reply .github/workflows/build.yml:12]: Sure, I'll fix it. Anyway, it passed on my repo with the same code https://github.com/hientranea/kotlin-quarkus-realworld-example-app/actions/runs/3174494671/jobs/5171391374 😂

**Review by u-ways — APPROVED** [review]

**Comment by u-ways** [comment]
@tranphuochien thanks! Sorry this ticket gave you more trouble than it should've, I think this seems to be a common issue with CI builds in GH actions. I will raise a ticket and link it here. (feel free to pick it up if interested)

**Comment by u-ways** [comment]
@tranphuochien looks like it was an issue with the workflow setup as GH actions PR already have write access, it failed on main and another branch, but I have resolved the issue now. You can see green runs here: https://github.com/u-ways/kotlin-quarkus-realworld-example-app/actions (#12) and latest workflow file can be found on main.

**Comment by hientranea** [comment]
Ohh TIL. Thank you 👍

**Pull request merged**